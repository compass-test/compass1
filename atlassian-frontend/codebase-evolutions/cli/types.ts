export type Flags = {
  readonly debug?: boolean;
};

export type ConfirmResponse = { answer: boolean };
export type InputResponse = { answer: string };
export type SelectResponse = { answer: string | string[] };

export type UserInputType = {
  task?: string;
  allDependencies?: string[];
  hasPostUpgradeTasks?: boolean;
  hasSingletons?: boolean;
  upgradeIdentifier?: string;
  rolloutStrategy?: string;
};

export type UserActionType = {
  key: string;
  question: string;
  answer: any;
};

export type Query = {
  // Used when asking the question, and in the summary
  question: string;
  // Only used when asking the question
  context?: string | string[];
};

export type Answer = string | string[] | boolean;

export type Choice = {
  value: string | string[];
  nextDecisionKey?: keyof UserInputType;
};

export type Question = {
  type: 'confirm' | 'select' | 'select-multi' | 'input';
  query: Query | string;
  // Choices presented when type is select
  choices?: Choice[];
};

export type Decision = {
  key: keyof UserInputType;
  question: Question;
  // User provided answer or choice
  answer?: Answer;
  nextKey: (this: Decision, answer: Answer) => keyof UserInputType | undefined;
  next?: () => Decision | undefined;
  prev?: () => Decision | undefined;
  setUserInput?: (userInput: UserInputType) => void;
};
