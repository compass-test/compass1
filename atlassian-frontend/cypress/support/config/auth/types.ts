export interface IdentityConfiguration {
  verifyHostname: string;
  loginHostName: string;
}

export interface CookieConfiguration {
  name: string;
  domain: string;
}

export interface AuthConfiguration {
  username: string;
  password: string;
  identity: IdentityConfiguration;
  cookie: CookieConfiguration;
}
