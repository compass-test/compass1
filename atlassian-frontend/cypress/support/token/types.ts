export interface IdentityVerifyResponse {
  token: string;
}
export interface TokenServiceOpts {
  identityHostname: string;
  identityLoginHostname: string;
  identityUsername: string;
  identityPassword: string;
  cookieName: string;
}

export interface UserSession {
  sessionToken: string;
  xsrfToken: string;
}
