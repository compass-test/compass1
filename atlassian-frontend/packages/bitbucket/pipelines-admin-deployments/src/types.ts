import { Environment } from '@atlassian/pipelines-models';

export type ReorderEnvironments = (
  type: string,
  environmentTypeOrder: { uuid: string }[],
) => void;

export type FetchBranches = (
  name: string,
) => Promise<{ values: { name: string }[] }>;

export type CreateBranchRestriction = (
  environmentUuid: string,
  pattern: string,
) => void;

export type DeleteBranchRestriction = (
  environmentUuid: string,
  restrictionUuid: string,
) => void;

export type CreateEnvironment = (
  environmentName: string,
  environmentType: string,
) => void;

export type UpdateEnvironment = (environmentUuid: string, change: any) => void;

export type DeleteEnvironment = (environmentUuid: string) => void;

export type RenderVariableList = (environment: Environment) => React.ReactNode;
