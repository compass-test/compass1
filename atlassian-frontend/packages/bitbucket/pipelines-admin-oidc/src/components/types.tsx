export type EnvironmentSelectItem = {
  label: string;
  value: string;
  uuid: string;
};

export type DeploymentEnvironment = {
  label: string;
  options: EnvironmentSelectItem[];
};
