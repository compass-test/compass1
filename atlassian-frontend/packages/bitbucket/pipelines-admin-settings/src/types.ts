export type Account = {
  hasFetchedUser: boolean;
  userHas2FaEnabled: boolean;
};

export type Capabilities = {
  hasFetchedCapabilities: boolean;
  pipelinesEnabled: boolean;
};

export type BuildConfiguration = {
  hasYmlFile: boolean;
  hasFetchedRawYmlFile: boolean;
};
