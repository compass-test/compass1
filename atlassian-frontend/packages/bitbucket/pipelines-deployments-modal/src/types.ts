export enum ModalType {
  PREVIEW = 'preview',
  SUMMARY = 'summary',
  REDEPLOY = 'redploy',
  STEP_SUMMARY = 'step_summary',
  STEP_PREVIEW = 'step_preview',
  STEP_REDEPLOY = 'step_redploy',
}

export type Line = {
  oldLine?: number;
  newLine?: number;
  type: 'normal' | 'add' | 'del' | 'empty' | 'loaded';
  content: string;
  wordDiff?: string;
  position?: number;
  conflictType: 'marker' | 'content';
};

export type ChunkEntry = {
  id: string;
  newStart: number;
  oldStart: number;
  newLines: number;
  oldLines: number;
  content: string;
  changes: Line[];
};

export type Diff = {
  chunks: ChunkEntry[];
  from: string;
  to: string;
};

export type DiffContent = {
  lineFrom?: number;
  lineTo?: number;
  content: string;
};

export type ChunkReducerAccumulator = {
  before: ChunkEntry;
  after: ChunkEntry;
};

export type ChangeReducerAccumulator = {
  before: Line[];
  after: Line[];
};

export type GroupChangeAccumulator = {
  loadedBefore: Line[];
  loadedAfter: Line[];
  rest: Line[];
};
