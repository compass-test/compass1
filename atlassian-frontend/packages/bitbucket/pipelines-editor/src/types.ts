import type { SupportedLanguages } from '@atlaskit/code/types';
import type { Variable } from '@atlassian/pipelines-variables';

export enum SyntaxType {
  Text,
  CodeBlock,
}

export interface PipeReadmeSyntax {
  type?: SyntaxType;
  html?: string;
  innerText?: string;
  language?: SupportedLanguages;
}

export type Step = {
  name: string;
  description: string;
  logo: any;
  yml: string;
};

export type Pipe = {
  category: string;
  name: string;
  description: string;
  logo: string;
  yml: string;
  repositoryPath?: string;
  version?: string;
  maintainer?: { name: string; website: string };
  vendor?: { name: string };
  timestamp?: number;
  tags: string[];
};

export type Template = {
  name: string;
  description: string;
  language: string;
  logo: string;
  yml: string;
};

export type VariablesProps = {
  createVariable: (variable: Variable, environmentUuid?: string) => void;
  deleteVariable: (variable: Variable, environmentUuid?: string) => void;
  updateVariable: (
    previousVariable: Variable,
    variable: Variable,
    environmentUuid?: string,
  ) => void;
  isFetchingVariables: boolean;
  isReadOnly: boolean;
  variables: Variable[];
  environments?: { [uuid: string]: string };
  environmentVariables?: { [uuid: string]: Variable[] };
};
