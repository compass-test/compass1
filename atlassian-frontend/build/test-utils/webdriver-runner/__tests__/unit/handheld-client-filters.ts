import { setBrowserStackMobileClients } from '../../lib/clients/setup-clients';
import {
  filterClientsByPlatform,
  filterClientsByPlatformVersion,
  filterClientsByFormFactor,
  filterClientsByKeyboard,
  ClientsFilter,
} from '../../lib/clients/mobile/client-filters';
import { MobileTestCaseOptions } from '../../types/';

function flattenClientToDeviceName(clients: any[]) {
  return clients.map(client => {
    const { os, os_version, device } = client.options.capabilities;
    return `${os} ${os_version} ${device}`;
  });
}

const clients = setBrowserStackMobileClients();

function snapshot(filterFunc: ClientsFilter, options: MobileTestCaseOptions) {
  const filtered = clients.filter(filterFunc(options));
  return flattenClientToDeviceName(filtered);
}

describe('BrowserStack', () => {
  describe('filterClientsByPlatform', () => {
    it('should skip entire platform', () => {
      const iosOnly = snapshot(filterClientsByPlatform, {
        skipPlatform: ['android'],
      });
      expect(iosOnly).toMatchSnapshot('iOS only');
      const androidOnly = snapshot(filterClientsByPlatform, {
        skipPlatform: ['ios'],
      });
      expect(androidOnly).toMatchSnapshot('Android only');
    });

    it('should skip all platforms', () => {
      const devices = snapshot(filterClientsByPlatform, {
        skipPlatform: ['android', 'ios'],
      });
      expect(devices.length).toBe(0);
    });

    it('should not skip when not used', () => {
      const emptyArray = snapshot(filterClientsByPlatform, {
        skipPlatform: [],
      });
      expect(emptyArray.length).toBe(clients.length);
      const explicitUndefined = snapshot(filterClientsByPlatform, {
        skipPlatform: undefined,
      });
      expect(explicitUndefined.length).toBe(clients.length);
      const undefinedParam = snapshot(filterClientsByPlatform, {});
      expect(undefinedParam.length).toBe(clients.length);
    });
  });

  describe('filterClientsByPlatformVersion', () => {
    it('should filter out the default version key when provided directly', () => {
      const filtered = snapshot(filterClientsByPlatformVersion, {
        versions: ['DEFAULT'],
      });
      expect(filtered.length).toBe(clients.length);
    });

    it('should target specific versions', () => {
      const filtered = snapshot(filterClientsByPlatformVersion, {
        versions: ['ios 12', 'android 9'],
      });
      expect(filtered).toMatchSnapshot();
    });

    it('should not skip when not used', () => {
      const emptyArray = snapshot(filterClientsByPlatformVersion, {
        versions: [],
      });
      expect(emptyArray.length).toBe(clients.length);
      const explicitUndefined = snapshot(filterClientsByPlatformVersion, {
        versions: undefined,
      });
      expect(explicitUndefined.length).toBe(clients.length);
      const undefinedParam = snapshot(filterClientsByPlatformVersion, {});
      expect(undefinedParam.length).toBe(clients.length);
    });
  });

  describe('filterClientsByFormFactor', () => {
    it('should target specific form factor devices', () => {
      const phonesOnly = snapshot(filterClientsByFormFactor, {
        formFactors: ['phone'],
      });
      expect(phonesOnly).toMatchSnapshot('Phones only');
      const tabletsOnly = snapshot(filterClientsByFormFactor, {
        formFactors: ['tablet'],
      });
      expect(tabletsOnly).toMatchSnapshot('Tablets only');
      const phonesAndTablets = snapshot(filterClientsByFormFactor, {
        formFactors: ['phone', 'tablet'],
      });
      expect(phonesAndTablets).toMatchSnapshot('Phones & Tablets');
    });

    it('should not skip when not used', () => {
      const emptyArray = snapshot(filterClientsByFormFactor, {
        formFactors: [],
      });
      expect(emptyArray.length).toBe(clients.length);
      const explicitUndefined = snapshot(filterClientsByFormFactor, {
        formFactors: undefined,
      });
      expect(explicitUndefined.length).toBe(clients.length);
      const undefinedParam = snapshot(filterClientsByFormFactor, {});
      expect(undefinedParam.length).toBe(clients.length);
    });
  });

  describe('filterClientsByKeyboard', () => {
    it('should target specific keyboards', () => {
      const samsung = snapshot(filterClientsByKeyboard, {
        keyboards: ['samsung'],
      });
      expect(samsung).toMatchSnapshot('Samsung only');
      const appleGoogle = snapshot(filterClientsByKeyboard, {
        keyboards: ['apple', 'gboard'],
      });
      expect(appleGoogle).toMatchSnapshot('Apple & Google');
    });

    it('should not skip when not used', () => {
      const emptyArray = snapshot(filterClientsByKeyboard, {
        keyboards: [],
      });
      expect(emptyArray.length).toBe(clients.length);
      const explicitUndefined = snapshot(filterClientsByKeyboard, {
        keyboards: undefined,
      });
      expect(explicitUndefined.length).toBe(clients.length);
      const undefinedParam = snapshot(filterClientsByKeyboard, {});
      expect(undefinedParam.length).toBe(clients.length);
    });
  });
});
