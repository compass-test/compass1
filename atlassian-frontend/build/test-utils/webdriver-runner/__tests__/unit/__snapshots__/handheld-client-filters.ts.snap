// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowserStack filterClientsByFormFactor should target specific form factor devices: Phones & Tablets 1`] = `
Array [
  "iOS 14 iPhone 12",
  "iOS 14 iPad Pro 12.9 2020",
  "iOS 13 iPhone 11 Pro",
  "iOS 13 iPad Air 2019",
  "iOS 12 iPhone XS",
  "iOS 12 iPad Air 2019",
  "Android 11.0 Google Pixel 4",
  "Android 10.0 Samsung Galaxy S20",
  "Android 10.0 Google Pixel 4",
  "Android 10.0 Samsung Galaxy Tab S10",
  "Android 9.0 Samsung Galaxy S10",
  "Android 9.0 Google Pixel 3",
  "Android 9.0 Samsung Galaxy Tab S6",
  "Android 8.0 Samsung Galaxy S9",
  "Android 8.0 Google Pixel 3",
  "Android 8.0 Samsung Galaxy Tab S4",
  "Android 7.0 Samsung Galaxy S8",
  "Android 7.0 Google Pixel",
  "Android 7.0 Samsung Galaxy Tab S3",
  "Android 6.0 Samsung Galaxy S7",
  "Android 6.0 Google Nexus 6",
]
`;

exports[`BrowserStack filterClientsByFormFactor should target specific form factor devices: Phones only 1`] = `
Array [
  "iOS 14 iPhone 12",
  "iOS 13 iPhone 11 Pro",
  "iOS 12 iPhone XS",
  "Android 11.0 Google Pixel 4",
  "Android 10.0 Samsung Galaxy S20",
  "Android 10.0 Google Pixel 4",
  "Android 9.0 Samsung Galaxy S10",
  "Android 9.0 Google Pixel 3",
  "Android 8.0 Samsung Galaxy S9",
  "Android 8.0 Google Pixel 3",
  "Android 7.0 Samsung Galaxy S8",
  "Android 7.0 Google Pixel",
  "Android 6.0 Samsung Galaxy S7",
  "Android 6.0 Google Nexus 6",
]
`;

exports[`BrowserStack filterClientsByFormFactor should target specific form factor devices: Tablets only 1`] = `
Array [
  "iOS 14 iPad Pro 12.9 2020",
  "iOS 13 iPad Air 2019",
  "iOS 12 iPad Air 2019",
  "Android 10.0 Samsung Galaxy Tab S10",
  "Android 9.0 Samsung Galaxy Tab S6",
  "Android 8.0 Samsung Galaxy Tab S4",
  "Android 7.0 Samsung Galaxy Tab S3",
]
`;

exports[`BrowserStack filterClientsByKeyboard should target specific keyboards: Apple & Google 1`] = `
Array [
  "iOS 14 iPhone 12",
  "iOS 14 iPad Pro 12.9 2020",
  "iOS 13 iPhone 11 Pro",
  "iOS 13 iPad Air 2019",
  "iOS 12 iPhone XS",
  "iOS 12 iPad Air 2019",
  "Android 11.0 Google Pixel 4",
  "Android 10.0 Google Pixel 4",
  "Android 9.0 Google Pixel 3",
  "Android 8.0 Google Pixel 3",
  "Android 7.0 Google Pixel",
  "Android 6.0 Google Nexus 6",
]
`;

exports[`BrowserStack filterClientsByKeyboard should target specific keyboards: Samsung only 1`] = `
Array [
  "Android 10.0 Samsung Galaxy S20",
  "Android 10.0 Samsung Galaxy Tab S10",
  "Android 9.0 Samsung Galaxy S10",
  "Android 9.0 Samsung Galaxy Tab S6",
  "Android 8.0 Samsung Galaxy S9",
  "Android 8.0 Samsung Galaxy Tab S4",
  "Android 7.0 Samsung Galaxy S8",
  "Android 7.0 Samsung Galaxy Tab S3",
  "Android 6.0 Samsung Galaxy S7",
]
`;

exports[`BrowserStack filterClientsByPlatform should skip entire platform: Android only 1`] = `
Array [
  "Android 11.0 Google Pixel 4",
  "Android 10.0 Samsung Galaxy S20",
  "Android 10.0 Google Pixel 4",
  "Android 10.0 Samsung Galaxy Tab S10",
  "Android 9.0 Samsung Galaxy S10",
  "Android 9.0 Google Pixel 3",
  "Android 9.0 Samsung Galaxy Tab S6",
  "Android 8.0 Samsung Galaxy S9",
  "Android 8.0 Google Pixel 3",
  "Android 8.0 Samsung Galaxy Tab S4",
  "Android 7.0 Samsung Galaxy S8",
  "Android 7.0 Google Pixel",
  "Android 7.0 Samsung Galaxy Tab S3",
  "Android 6.0 Samsung Galaxy S7",
  "Android 6.0 Google Nexus 6",
]
`;

exports[`BrowserStack filterClientsByPlatform should skip entire platform: iOS only 1`] = `
Array [
  "iOS 14 iPhone 12",
  "iOS 14 iPad Pro 12.9 2020",
  "iOS 13 iPhone 11 Pro",
  "iOS 13 iPad Air 2019",
  "iOS 12 iPhone XS",
  "iOS 12 iPad Air 2019",
]
`;

exports[`BrowserStack filterClientsByPlatformVersion should target specific versions 1`] = `
Array [
  "iOS 12 iPhone XS",
  "iOS 12 iPad Air 2019",
  "Android 9.0 Samsung Galaxy S10",
  "Android 9.0 Google Pixel 3",
  "Android 9.0 Samsung Galaxy Tab S6",
]
`;
